extendr-api/src/wrapper/mod.rs

```diff
@@ -102,42 +102,59 @@ macro_rules! make_conversions {
 make_conversions!(Pairlist, ExpectedPairlist, is_pairlist, "Not a pairlist");
 make_conversions!(Function, ExpectedFunction, is_function, "Not a function");
 make_conversions!(Raw, ExpectedRaw, is_raw, "Not a raw object");
+
 make_conversions!(
     Character,
     ExpectedCharacter,
     is_character,
     "Not a character object"
 );
+
 make_conversions!(
     Environment,
     ExpectedEnviroment,
     is_environment,
     "Not an Environment"
 );
 
+make_conversions!(List, ExpectedList, is_list, "Not a List");
+
+make_conversions!(
+    Expression,
+    ExpectedExpression,
+    is_expression,
+    "Not an Expression"
```

Comment:
We should be consistent with capitalization here. Compare "Not an Expression" to "Not a function".  I would probably use lower case throughout, but upper case is fine if it's consistently used.

Reply:
It has been a bit of a whirl...

Thanks for spotting these typos.

I would love to drop `FromRobj` entirely in the future and convert everything to `TryFrom`
using the formal Error. The error would implement `Display` to generate the error message,

The essence is that From and `TryFrom` handle all the input and output cases.

The `Deref` means that you can use the wrapper as a `Robj`.

Rust doesn't have inheritance, per-se, but `Deref` allows you to access the `Base class`
which in this case is the wrapped Robj.

You should now be able to use just `Function`, `Environment`, `Symbol` etc. as parameter
types. These types provide access to the underlying object.

I plan to move many methods from Robj to their relevant wrappers.

The real test will be how easy it is for library writers to get to grips with this.